src/features/walletUnlocker/OnboardingPairing.tsx

```diff
@@ -40,25 +40,52 @@ export const OnboardingPairing = (): JSX.Element => {
   const navigate = useNavigate();
   const dispatch = useTypedDispatch();
   const { useProxy, isTauri } = useTypedSelector(({ settings }) => settings);
+  const isConnectUrlFromEnvVar = useTypedSelector(
+    ({ walletUnlocker }) => walletUnlocker.isConnectUrlFromEnvVar
+  );
 
   useEffect(() => {
-    // Reset the APIs state completely
-    // Especially useful when redirected on this page after failed pairing attempt
-    dispatch(liquidApi.util.resetApiState());
-    dispatch(operatorApi.util.resetApiState());
-    dispatch(walletUnlockerApi.util.resetApiState());
-    dispatch(walletApi.util.resetApiState());
+    (async () => {
+      // Reset the APIs state completely
+      // Especially useful when redirected on this page after failed pairing attempt
+      dispatch(liquidApi.util.resetApiState());
+      dispatch(operatorApi.util.resetApiState());
+      dispatch(walletUnlockerApi.util.resetApiState());
+      dispatch(walletApi.util.resetApiState());
+      // Auto connect if tdex connect url is in env var (like in Umbrel)
+      if (process.env.REACT_APP_TDEX_CONNECT_URL) {
+        // isConnectUrlFromEnvVar is false on first load
+        if (!isConnectUrlFromEnvVar) {
+          dispatch(setIsConnectUrlFromEnvVar(true));
+          await onFinish(process.env.REACT_APP_TDEX_CONNECT_URL);
```

Comment:
this will not be avaialble at runtime, we need the window DOM trick

https://github.com/tdex-network/tdex-dashboard/blob/master/web-entrypoint.sh